docs/DeveloperGuide.md

```diff
@@ -308,32 +308,197 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 ### Use cases
 
-(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
+(For all use cases below, the **System** is the `NetConnect` and the **Actor** is the `user`, unless specified otherwise)
 
-**Use case: Delete a person**
+**Use case: UC01 - List All Persons**
 
 **MSS**
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1. User requests to list all persons.
+2. NetConnect shows the list of all persons.
 
     Use case ends.
 
 **Extensions**
 
-* 2a. The list is empty.
+* 2a. The list of all persons is empty.
 
   Use case ends.
 
-* 3a. The given index is invalid.
+**Use case: UC02 - Find a List of Persons by Name**
 
-    * 3a1. AddressBook shows an error message.
+**MSS**
+
+1. User requests for the list of persons matching a name.
+2. NetConnect shows a list of persons with matching name.
+
+   Use case ends.
+
+**Extensions**
+
+* 2a. There are no persons with a matching name.
+
+  Use case ends.
+
+**Use case: UC03 - Find a Specific Person by Contact Number**
+
+**MSS**
+
+1. User requests for the person with the matching contact number.
+2. NetConnect shows the person with matching contact number.
+
+   Use case ends.
```

Comment:
Invalid numbers and returning an input mistake could be an Extension to Use Case 03.